piano-hero/mobile/src/main/java/com/novoda/pianohero/SimpleEndToEndActivity.java

```diff
@@ -60,35 +34,24 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == R.id.show_keyboard) {
-            showKeyboard();
+            presenter.onShowKeyboardSelected();
             return true;
         } else if (item.getItemId() == R.id.restart_game) {
-            startNewGame();
+            presenter.onRestartGameSelected();
             return true;
         } else {
             return super.onOptionsItemSelected(item);
         }
     }
 
-    private void showKeyboard() {
-        InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
-        inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY);
-    }
-
     @Override
     public boolean onKeyUp(int keyCode, KeyEvent event) {
         try {
             Note note = keyCodeConverter.convert(keyCode);
-            brain.onNotesPlayed(note);
+            presenter.onNotesPlayed(note);
         } catch (IllegalArgumentException e) {
-            PianoHeroApplication.popToast("that's not a simple note!");
+            Log.e("!!!", "Non supported key input");
```

Comment:
you cross it out, but just to confirm - no user feedback because we never expect to allow non supported input